planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java

```diff
@@ -108,10 +108,7 @@ public class Place implements
   private static final double MAX_CITY_POPULATION = 100_000_000d;
   private static final Set<String> MAJOR_CITY_PLACES = Set.of("city", "town", "village");
   private static final ZoomFunction<Number> LABEL_GRID_LIMITS = ZoomFunction.fromMaxZoomThresholds(Map.of(
-    8, 4,
-    9, 8,
-    10, 12,
-    12, 14
+    12, 15
```

Comment:
I don't understand what's going on here, can you explain it?

Reply:
Replicates https://github.com/ACMEAtronOmatic/openmaptiles/pull/1/commits/020cc15aefdeedb700013c9d9841efa4895c3fb5

Explanation:
This function maps zoom levels to 'label limits'. The label limits are used to determine how many labels are allowed to appear in a given square of the tile per zoom (the tile is split into a grid). This only applies to places that do not have a rank, i.e. are not found in Natural Earth data. See line https://github.com/nmr8acme/planetiler/pull/1/files#diff-8e85b00e58d492c0d11ae5481a17f671b0831a1c41c157c695b01a7749f06a5fR353 

By deleting the limits up to zoom 12, the limit of number of place labels per square on the grid is increased, so more places show. In the SQL version this was achieved by assigning a 'gridrank' number to each place.

It took me a while to figure out what was happening here, hopefully that makes sense